File: java/src/main/java/codingdojo/CustomerDataLayer.java
package codingdojo;

public interface CustomerDataLayer {

    Customer updateCustomerRecord(Customer customer);

    Customer createCustomerRecord(Customer customer);

    void updateShoppingList(ShoppingList consumerShoppingList);

    Customer findByExternalId(String externalId);

    Customer findByMasterExternalId(String externalId);

    Customer findByCompanyNumber(String companyNumber);
}

File: typescript/src/CustomerDataLayer.ts
import {Customer} from "@/Customer";
import {ShoppingList} from "@/ShoppingList";

export interface CustomerDataLayer {
  updateCustomerRecord(customer: Customer): Promise<Customer | null>;
  createCustomerRecord(customer: Customer): Promise<Customer>;
  updateShoppingList(consumerShoppingList: ShoppingList): Promise<void>;
  findByExternalId(externalId: string): Promise<Customer | null>;
  findByMasterExternalId(externalId: string): Promise<Customer | null>;
  findByCompanyNumber(companyNumber: string | null): Promise<Customer | null>;
}
